types/pino/index.d.ts

```diff
@@ -444,6 +444,11 @@ declare namespace P {
          * Flushes the content of the buffer in extreme mode. It has no effect if extreme mode is not enabled.
          */
         flush(): void;
+
+        /**
+         * A utility method for determining if a given log level will write to the destination.
+         */
+        isLevelEnabled(level: LevelWithSilent | string): boolean;
```

Comment:
Why a generic string? If you define `LevelWithSilent | string` you could get rid of `LevelWithSilent` because every string is valid and I guess it should only be allowed to set allowed levels. So I would delete `string`

Reply:
Ok, I saw that you can set a level with LevelWithSilent | string but in my opinion this is wrong as well. If you define union for string literals and a generic string the string literal makes no sense anymore.

Reply:
My best guess is that this could potentially provide hints to documentation-generators and IDE autocompletions.

This convention is followed in 4 other places in this definitions file, so I'd be a little hesitant to depart from it here without a good reason.

Reply:
Well, actually it does. You can use the `LiteralUnion` type from https://github.com/sindresorhus/type-fest to get autocompletion for such union types in an IDE.

Reply:
It's actually more of a bug in TypeScript that it doesn't work.

Reply:
All right, good to know :+1:

Reply:
@screendriver Are you comfortable approving this PR?

Reply:
I thought it would be enough that @BendingBender approved it. I approved it also now.